# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/TextUpdater.cpp
# Commit: a81a40b250b4
# Full Hash: a81a40b250b401d6133c2490d8d2d3bd52f1bc6d
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-05-05 09:42:30
# Regressor Bug: 1766794
# File Overlap Count: 2
# Description:
#   Bug 1766794: Lazily cache HyperText offsets for RemoteAccessibles. r=eeejay
#   
#   We already had a similar cache in local HyperTextAccessible.
#   This improves performance significantly when walking the text attributes of a container with a large number of text leaf children, such as is encountered when using view source.
#   This patch unifies that cache across local and remote.
# ==============================================================================

diff -r 000ea1907a05 -r a81a40b250b4 accessible/base/TextUpdater.cpp
--- a/accessible/base/TextUpdater.cpp	Thu May 05 04:00:58 2022 +0000
+++ b/accessible/base/TextUpdater.cpp	Thu May 05 04:31:12 2022 +0000
@@ -44,8 +44,7 @@
     return;
   }
 
-  // Get the text leaf accessible offset and invalidate cached offsets after it.
-  mTextOffset = mHyperText->GetChildOffset(mTextLeaf, true);
+  mTextOffset = mHyperText->GetChildOffset(mTextLeaf);
   NS_ASSERTION(mTextOffset != -1, "Text leaf hasn't offset within hyper text!");
 
   uint32_t oldLen = aOldText.Length(), newLen = aNewText.Length();
@@ -90,6 +89,7 @@
 
     // Update the text.
     mTextLeaf->SetText(aNewText);
+    mHyperText->InvalidateCachedHyperTextOffsets();
     return;
   }
 
@@ -135,6 +135,7 @@
 
   // Update the text.
   mTextLeaf->SetText(aNewText);
+  mHyperText->InvalidateCachedHyperTextOffsets();
 }
 
 void TextUpdater::ComputeTextChangeEvents(